documentation/docs/decentralized_identifiers/resolve.mdx

```diff
@@ -7,18 +7,152 @@ keywords:
+The [IOTA Identity Framework](https://github.com/iotaledger/identity.rs) supports resolving DID Documents that are stored on an IOTA Tangle (public or private). The main tool supplied 
+by the IOTA Identity Framework to handle DID Document resolution in a type safe manner is the `Resolver`. A DID Resolver as defined in the [W3C Decentralized Identifiers specification](https://www.w3.org/TR/did-core/#dfn-did-resolvers)
+enforces the signature of the resolution function in a manner that is more centered around Web/API resolution rather than a strongly typed framework. This is the reason why the `Resolver` provided by the IOTA Identity Framework deviates somewhat from 
+the W3C specification.
 
-A short example that shows how you can resolve a DID Document. This example returns the latest DID Document.
 
-<CodeSnippet 
-    nodeReplitLink="https://repl.it/@IOTAFoundation/resolution?lite=true"
-    rustContent={resolution}
-    nodeGithubLink = "https://github.com/iotaledger/identity.rs/blob/support/v0.5/bindings/wasm/examples/src/resolution.js"
-    rustGithubLink = "https://github.com/iotaledger/identity.rs/blob/support/v0.5/examples/low-level-api/resolution.rs"
-/>
+## Resolving a DID from the main network
+
+The following example demonstrates how to resolve the DID: "did:iota:H3C2AVvLMv6gmMNam3uVAjZpfkcJCwDwnZn6z3wXmqPV" from the `main` network. 
+<Tabs>
+<TabItem value="Rust" label="Rust">
+
+```rust
+use identity::iota::Resolver; 
+use identity::iota_core::IotaDID;
+use identity::iota::ResolvedIotaDocument; 
+
+let resolver: Resolver = Resolver::new().await?; 
+let did: IotaDID = IotaDID::parse("did:iota:H3C2AVvLMv6gmMNam3uVAjZpfkcJCwDwnZn6z3wXmqPV")?;
+
+let doc: ResolvedIotaDocument = resolver.resolve(&did).await?; 
+
+```
+
+</TabItem>
+<TabItem value="node" label="Node.js">
+
+```js
+const {
+    DID,
+    Resolver,
+    ResolvedDocument,
+} = require('@iota/identity-wasm/node');
+
+ const resolver = new Resolver();
+ const did = DID.parse("did:iota:H3C2AVvLMv6gmMNam3uVAjZpfkcJCwDwnZn6z3wXmqPV");
+ const doc = await resolver.resolve(did);
+
+```
+</TabItem>
+</Tabs>
+The technical details of how resolution exactly works in terms of querying the Tangle is explained in the [IOTA DID Method Specification](./../specs/did/iota_did_method_spec#read). 
+
+## Resolving from a private tangle 
+Resolving a DID from a private tangle is similar to resolving a DID from the main net. The only difference is that 
+the resolver needs to be configured to have a client capable of operating on said private tangle. Building a `Client` configured for a specified Tangle is explained in [this example in Rust](https://github.com/iotaledger/identity.rs/blob/dev/examples/low-level-api/private_tangle.rs) and [this example in Javascript](https://github.com/iotaledger/identity.rs/blob/dev/bindings/wasm/examples/src/private_tangle.js).
```

Comment:
Direct links to the dev branch are risky for examples as a single chance would break this and doesn't help with future versioning of the wiki. We need to consider what to do for this kind of cases (@eike-hass relevant for you as well I think)